darshan-util/pydarshan/darshan/experimental/plots/plot_posix_access_pattern.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+def autolabel(ax, rects):
+    """Attach a text label above each bar in *rects*, displaying its value."""
+    for rect in rects:
+        height = rect.get_height()
+        ax.annotate(
+            '{}'.format(height),
+            xy=(rect.get_x() + rect.get_width() / 2, height),
+            xytext=(0, 3),  # 3 points vertical offset
+            textcoords="offset points",
+            ha='center',
+            va='bottom',
+            rotation=0,
+        )
+
+def plot_posix_access_pattern(record, ax=None):
+    """
+    Plots read/write access patterns (sequential vs consecutive access counts)
+    for a given POSIX module file record.
+
+	Args:
+		record (dict): POSIX module record to plot access pattern for.
```

Comment:
> This is the first plotting routine that takes a dataframe as input 

This is also a bit confusing for public consumption--it is actually a dictionary with the single-row data frames and other stuff nested inside of it. That will probably need to be crystal clear for end users eventually.

Reply:
I'm going to open a separate issue to discuss potential reworking of plots to and what might make sense for different routines to take as input, and hopefully we can clarify stuff like this there.

The intention is to allow users to pass in an individual record to this routine to have data from it plotted -- that record could represent a single file or, as in the case of the job summary tool, an accumulated record of all files accessed by the application.